### PR TITLE
[WNMGDS-2232] Merge CSS-Reset epic

### PR DESCRIPTION
**Pull request by pwolfert**
## Summary

In order to [prepare for the next beta release](https://jira.cms.gov/browse/WNMGDS-2232), merge the epic branch for the CSS Reset.

## How to test

[Here's a link to the diff without VRT snapshots](https://github.com/CMSgov/design-system/pull/2460/files?file-filters%5B%5D=.css&file-filters%5B%5D=.js&file-filters%5B%5D=.jsx&file-filters%5B%5D=.mdx&file-filters%5B%5D=.scss&file-filters%5B%5D=.snap&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx&show-viewed-files=true#) if you want to give the whole thing another look through. The diff with all the images is too large and overloads the page for me.

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [x] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [x] Selected appropriate `Impacts`, multiple can be selected.

### If this is a change to code:

- [x] Created or updated unit tests to cover any new or modified code
- [x] If necessary, updated unit-test snapshots (`yarn test:unit:update`) and browser-test snapshots (`yarn test:browser:update`)

### If this is a change to documentation:

- [x] Checked for spelling and grammatical errors

### CODE REVIEW

**Comment by forrestbaer** [comment]
Two somewhat major visual differences I'm noticing in the privacy settings dialog and the healthcare footer:

image
image

Will see if I can troubleshoot.

**Comment by forrestbaer** [comment]
The footer change is due to `[data-theme=healthcare] ol, [data-theme=healthcare] ul` setting `flex-direction: column` and could be fixed by changing this `ul` to `flex-direction: row`

**Comment by forrestbaer** [comment]
The privacy settings change looks ok it's just shifted down slightly, not sure exactly why yet, the `border-top` in this component doesn't look very good though and causes that text to look like it's bumping right up against the line.

**Comment by forrestbaer** [comment]
Looks like the privacy settings table is just a `ds-c-dialog__body p` issue with all `p` tags having `margin-block: 1em 0` and this first paragraph should be something like `margin-block: 1em 1em`

File: tests/browser/snapshots/components-choicelist--inverse-option-healthcare--Mobile-Chrome.png
Comment (zarahzachz) [inline tests/browser/snapshots/components-choicelist--inverse-option-healthcare--Mobile-Chrome.png:1]: Should choice 2 have a red border?

File: tests/browser/snapshots/components-drawer--drawer-default-cmsgov--Mobile-Chrome.png
Comment (zarahzachz) [inline tests/browser/snapshots/components-drawer--drawer-default-cmsgov--Mobile-Chrome.png:1]: Should this heading be bold?

File: tests/browser/snapshots/components-drawer--drawer-default-core--Mobile-Chrome.png
Comment (zarahzachz) [inline tests/browser/snapshots/components-drawer--drawer-default-core--Mobile-Chrome.png:1]: Core had bold heading, so perhaps this is a regression

**Comment by pwolfert** [comment]
> The footer change is due to `[data-theme=healthcare] ol, [data-theme=healthcare] ul` setting `flex-direction: column` and could be fixed by changing this `ul` to `flex-direction: row`

I thought I had fixed the footer flex direction issue in [4846577](https://github.com/CMSgov/design-system/pull/2460/commits/4846577ca860591c8644dab16c9befdce3a84fc3) 🤔

File: tests/browser/snapshots/components-privacysettingsdialog--default-cmsgov--Mobile-Chrome.png
Comment (zarahzachz) [inline tests/browser/snapshots/components-privacysettingsdialog--default-cmsgov--Mobile-Chrome.png:1]: Spacing issues here - probably from removing the bottom margin?

File: tests/browser/snapshots/components-privacysettingsdialog--default-cmsgov--chromium.png
Comment (zarahzachz) [inline tests/browser/snapshots/components-privacysettingsdialog--default-cmsgov--chromium.png:1]: Weird text flow on the choice labels here.

Comment (pwolfert) [reply tests/browser/snapshots/components-privacysettingsdialog--default-cmsgov--chromium.png:1]: Clicking through crashes the page for me. Was this the thing fixed by increasing the min width?

File: tests/browser/snapshots/foundations-typography-headings--responsive-utility-headings-cmsgov--Mobile-Chrome.png
Comment (zarahzachz) [inline tests/browser/snapshots/foundations-typography-headings--responsive-utility-headings-cmsgov--Mobile-Chrome.png:1]: This should be bold

File: tests/browser/snapshots/healthcare-footer--basic-footer-core--Mobile-Chrome.png
Comment (zarahzachz) [inline tests/browser/snapshots/healthcare-footer--basic-footer-core--Mobile-Chrome.png:1]: :suspect: 

(Is that the dude from Wolfenstein's Castle?)

Comment (zarahzachz) [reply tests/browser/snapshots/healthcare-footer--basic-footer-core--Mobile-Chrome.png:1]: These hgov footer images look like they should be removed.

Comment (zarahzachz) [reply tests/browser/snapshots/healthcare-footer--basic-footer-core--Mobile-Chrome.png:1]: Oh wait... it takes screenshots of hgov's footer using the Core theme. Weird and not helpful

File: tests/browser/snapshots/healthcare-header--minimal-header-healthcare--webkit.png
Comment (zarahzachz) [inline tests/browser/snapshots/healthcare-header--minimal-header-healthcare--webkit.png:1]: did the spacing of the logo change?

File: tests/browser/snapshots/components-privacysettingsdialog--default-cmsgov--chromium.png
Comment (zarahzachz) [reply tests/browser/snapshots/components-privacysettingsdialog--default-cmsgov--chromium.png:1]: I pulled the latest from this branch and it appears fixed.

File: tests/browser/snapshots/healthcare-header--minimal-header-healthcare--webkit.png
Comment (zarahzachz) [reply tests/browser/snapshots/healthcare-header--minimal-header-healthcare--webkit.png:1]: Maybe it didn't? Having a hard time seeing if this is a real issue. When looking at the "one column page layout" Story, the header logo aligns with the logo in the footer, so I assume the spacing is correct here.

File: tests/browser/snapshots/components-choicelist--inverse-option-healthcare--Mobile-Chrome.png
Comment (zarahzachz) [reply tests/browser/snapshots/components-choicelist--inverse-option-healthcare--Mobile-Chrome.png:1]: Brought up in standup (WONT FIX)

File: tests/browser/snapshots/foundations-typography-headings--responsive-utility-headings-cmsgov--Mobile-Chrome.png
Comment (zarahzachz) [reply tests/browser/snapshots/foundations-typography-headings--responsive-utility-headings-cmsgov--Mobile-Chrome.png:1]: Nevermind, just noticed some of our headings have different weights.

**Review by forrestbaer — APPROVED** [review]

**Review by zarahzachz — APPROVED** [review]

**Pull request merged**